File: src/main/java/io/github/hdzitao/editstarters/cache/MemoryCache.java
package io.github.hdzitao.editstarters.cache;

import java.util.WeakHashMap;
import java.util.function.Function;

/**
 * 内存缓存
 *
 * @version 3.2.0
 */
public class MemoryCache<K, V> {
    private final WeakHashMap<K, V> cache = new WeakHashMap<>();
    private final Function<K, V> transFun;


    public MemoryCache(Function<K, V> transFun) {
        this.transFun = transFun;
    }

    public V get(K k) {
        return cache.computeIfAbsent(k, transFun);
    }
}

File: src/main/kotlin/io/github/hdzitao/editstarters/cache/MemoryCache.kt
package io.github.hdzitao.editstarters.cache

import java.util.*
import java.util.function.Function

/**
 * 内存缓存
 *
 * @version 3.2.0
 */
class MemoryCache<K, V>(private val transFun: Function<K, V>) {
    private val cache = WeakHashMap<K, V>()

    fun get(k: K): V = cache.computeIfAbsent(k, transFun)
}
